### ISSUE TITLE
Sites Management Dashboard: Better abstraction for `isComingSoon`

### ISSUE THREAD

**Issue by danielbachhuber** [Automattic/wp-calypso#65411]
In #65359, I copy and pasted this bit of code for determining whether a site `isComingSoon`:

https://github.com/Automattic/wp-calypso/blob/03c84dd2376999ab07b6e6a74c4c3fbb22a8d282/client/sites-dashboard/components/sites-table-row.tsx#L73-L74

The code originated from `SitesTableFilterTabs`:

https://github.com/Automattic/wp-calypso/blob/d1fdb9ac04c87b6b2c83c5d39b647fa9961c84a5/client/sites-dashboard/components/sites-table-filter-tabs.tsx#L82-L83

I imagine future files will need access to this data. Ideally, we wouldn't copy and paste this logic between files.

The code originated with @p-jackson in #65182, so he might know why we need `site.is_coming_soon || ( site.is_private && site.launch_status === 'unlaunched' )` and not solely `site.is_coming_soon`.

**Done is:**
* We have a clean abstraction for determining whether a site is 'Launched', 'Coming Soon', or 'Private'.
* The abstraction is applied to our existing instances of the conditional logic.

**Comment by wojtekn** [comment]
@danielbachhuber I've checked other parts of Calypso for those usages, and we don't have a consistent way of using such checks. However, I spot the use of `isPrivateAndUnlaunched`. What if we refactor the current `isComingSoon` flag to use state selectors from `client/state/selectors`?:

```
const isPrivateAndUnlaunched = isPrivateSite && isUnlaunchedSite;
const isComingSoon = isSiteComingSoon || isPrivateAndUnlaunched; 
```

This looks much clearer.

Or alternatively, we could use only `isSiteComingSoon` state selector and move check for `isPrivateAndUnlaunched` there. However, we use `isSiteComingSoon` in four components, so we would need to validate the change there.

**Comment by p-jackson** [comment]
I believe @zaguiini is working on this in the PR that implements tile view #66325. Any time you need to decide which label to display on your site you can use an abstraction like this: https://github.com/Automattic/wp-calypso/pull/66325/files#diff-8bbf39155479aa6f51707f9e142fa24a88fe1f963b02b255a9f010bccce741faR11

**Comment by wojtekn** [comment]
@p-jackson As far as I see, the linked PR refactors label, but it doesn't refactor the `isComingSoon`. Will we extract this check there? We use this check not only to display the label but also for filtering.

**Comment by p-jackson** [comment]
That's true, it's used for filtering too 🤔 
Yeah we probably need it all to be using the one shared function.

**Comment by p-jackson** [comment]
Closed by #66511
